src/docs/release/breaking-changes/rendereditable-layout-before-hit-test.md

```diff
@@ -0,0 +1,84 @@
+---
+title: The RenderEditable needs to be laid out before hit testing
+description: The hit testing of RenderEditable requires additional information that is only available after the layout.
+---
+
+## Summary
+
+The `RenderEditable` needs to be laid out in order to process hit
+testing; Otherwise, the framework throws an assertion error indicates
+the `RenderEditable` has not been laid out.
+`Failed assertion: line xxxx pos xx: '!debugNeedsLayout': is not true.`
+
+## Context
+
+To support gesture recognizers in selectable text, the
+`RenderEditable` requires the layout information about the
+text spans to determine which text span receive the
+pointer event. Therefore, we made the layout to be a hard
+requirement for performing the hit testing to the
+`RenderEditable`.
+
```

Comment:
also add something like "In practice, this is rarely an issue. The widget library ensures that layout is performed before any hit test on all render objects. This problem is only likely to be seen in code that directly interacts with render objects, for example in tests of custom render objects."